addons/a11y/src/components/Report/HighlightToggle.tsx

```diff
@@ -28,106 +49,96 @@ function getElementBySelectorPath(elementPath: string): HTMLElement {
-    ownProps.elementsToHighlight,
+  const checkBoxState = areAllRequiredElementsHighlighted(
+    ownProps.elementsToHighlight || [],
     state.highlightedElementsMap
   );
   return {
     highlightedElementsMap: state.highlightedElementsMap,
-    isToggledOn,
+    isToggledOn: checkBoxState === CheckBoxStates.CHECKED,
+    indeterminate: checkBoxState === CheckBoxStates.INDETERMINATE,
   };
 };
 
-class HighlightToggle extends Component<ToggleProps, {}> {
+class HighlightToggle extends Component<ToggleProps> {
   static defaultProps: Partial<ToggleProps> = {
     elementsToHighlight: [],
   };
 
+  private checkBoxRef = React.createRef<HTMLInputElement>();
+
   componentDidMount() {
-    for (let element of this.props.elementsToHighlight) {
-      if (element) {
-        const targetElement = getElementBySelectorPath(element.target[0]);
-        if (targetElement && !this.props.highlightedElementsMap.get(targetElement)) {
-          this.saveElementDataToMap(
-            targetElement,
-            false,
-            targetElement.style.outline,
-            this.props.type
-          );
-        }
+    this.props.elementsToHighlight.forEach(element => {
+      const targetElement = getElementBySelectorPath(element.target[0]);
+      if (targetElement && !this.props.highlightedElementsMap.has(targetElement)) {
+        this.saveElementDataToMap(targetElement, false, targetElement.style.outline);
       }
+    });
+  }
+
+  componentDidUpdate(prevProps: Readonly<ToggleProps>): void {
+    if (this.checkBoxRef.current) {
+      this.checkBoxRef.current.indeterminate = this.props.indeterminate;
     }
   }
 
-  higlightRuleLocation(targetElement: HTMLElement, addHighlight: boolean): void {
-    const OUTLINE_STYLE = `dotted`;
-    const OUTLINE_WIDTH = `1px`;
-    if (targetElement) {
-      if (addHighlight) {
-        switch (this.props.type) {
-          case RuleType.PASS:
-            this.setTargetElementOutlineStyle(targetElement, `${convert(themes.normal).color.positive} ${OUTLINE_STYLE} ${OUTLINE_WIDTH}`);
-            break;
-          case RuleType.VIOLATION:
-            this.setTargetElementOutlineStyle(targetElement, `${convert(themes.normal).color.negative} ${OUTLINE_STYLE} ${OUTLINE_WIDTH}`);
-            break;
-          case RuleType.INCOMPLETION:
-            this.setTargetElementOutlineStyle(targetElement, `${convert(themes.normal).color.warning} ${OUTLINE_STYLE} ${OUTLINE_WIDTH}`);
-            break;
-        }
-      } else {
-        if (this.props.highlightedElementsMap.get(targetElement)) {
-          this.setTargetElementOutlineStyle(
-            targetElement,
-            this.props.highlightedElementsMap.get(targetElement).originalOutline
-          );
-        }
-      }
+  highlightRuleLocation(targetElement: HTMLElement, addHighlight: boolean): void {
+    if (!targetElement) {
+      return;
+    }
+
+    if (addHighlight) {
+      highlightElement(targetElement, this.props.type);
```

Comment:
I like what youre doing here but why dont you use one line: setElementOutlineStyle(targetElement, `${colorsByType[type]} dotted 1px`); rather than call a function that calls that one line of code

Reply:
Good catch. Thank you! Fixed.